doc/tutorial/sequence_file/sequence_file_solution4.cpp

```diff
@@ -0,0 +1,55 @@
+#include <fstream>
+
+struct write_file_dummy_struct
+{
+    write_file_dummy_struct()
+    {
+
+auto file_raw = R"//![fastq_file](
+@seq1
+CGATCGATC
++
+IIIIII!!!
+@seq2
+AGCGATCGAGGAATATAT
++
+IIIIHHGIII!!!!!!!!
+@seq3
+AGCTAGCAGCGATCG
++
+IIIIIHII!!!!!!!
+)//![fastq_file]";
+
+        std::ofstream file{"/tmp/my.fastq"};
+        std::string str{file_raw};
+        file << str.substr(1); // skip first newline
+    }
+};
+
+write_file_dummy_struct go{};
+
+//![solution]
+#include <seqan3/io/sequence_file/all.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/take_until.hpp>  // view::take_until
+
+using namespace seqan3;
+
+int main()
+{
+    sequence_file_input fin{"/tmp/my.fastq", fields<field::ID, field::SEQ_QUAL>{}};
+    sequence_file_output fout{"/tmp/trimmed.fastq", fields<field::ID, field::SEQ_QUAL>{}};
+
+    auto trimming = view::transform([] (auto & rec)
+    {
+        get<field::SEQ_QUAL>(rec) = (get<field::SEQ_QUAL>(rec)
+                                  | view::take_until([] (auto chr) { return chr.to_phred() <= 10; }));
```

Comment:
does `|=` work?

Reply:
in any case you don't need paranthesis around the rhs

Reply:
> does |= work?

No :( it assumes a bitwise operation.